packages/terra-clinical-header/src/Header.jsx

```diff
@@ -35,34 +53,53 @@ const propTypes = {
    * A Boolean indicating if element is a subheader.
    */
   isSubheader: PropTypes.bool,
+
+  /**
+   * IDs should be used with 'aria-labelledby' to associate headings with the corresponding page area when needing to label regions.
+   *
+   * For an example, read [Accessibility Guide: Labeling Section Content](/components/terra-clinical-header/clinical-header/accessibility-guide#labeling-section-content) for additional information.
+   */
+  id: PropTypes.string,
 };
 
 const defaultProps = {
   title: '',
   startContent: null,
   endContent: null,
   isSubheader: false,
+  text: '',
 };
 
 const Header = ({
-  children, title, startContent, endContent, isSubheader, ...customProps
+  children, title, startContent, endContent, text, level, id, isSubheader, ...customProps
 }) => {
   const theme = useContext(ThemeContext);
+  if (title) {
+    // TODO: remove on the next major version bump
+    // eslint-disable-next-line no-console
+    console.warn('The `title` prop has been renamed to `text`. Please update all references of `title` prop to `text`.');
+  }
+  if (!level) {
+    // TODO: remove on the next major version bump
+    // eslint-disable-next-line no-console
+    console.warn('Default heading level may not appropriate has it would fail to convey context of heading in a site / application where it is used. Heading level should be set explicitly depending on the position of header in site / application to allow screen readers to identify headers consistently.');
+  }
 
   let titleElement;
-  if (title) {
+  if (title || text) {
+    const HeaderElement = (level) ? `h${level}` : 'h1';
     titleElement = (
       <div className={cx('title-container')}>
-        <h1 className={cx('title')}>
-          {title}
-        </h1>
+        <HeaderElement id={id} className={cx('title')}>
+          {title || text}
+        </HeaderElement>
       </div>
     );
   }
 
   const content = React.Children.map(children, child => (
     React.cloneElement(child, {
-      className: classNames([cx('flex-collapse'), children.props.className]),
+      className: classNames([cx('flex-collapse'), child.props.className]),
```

Comment:
why is `children` changed to `child`?

Reply:
this was failing when we had multiple nodes / elements wrapped within clinical-header. children becomes array when it has more than one child items in this scenario it fail to retrieve class-names